components/Dialogs/examples/DialogsThemedPresentationExampleViewController.swift

```diff
@@ -0,0 +1,202 @@
+    }
+    set {
+      super.preferredContentSize = newValue
+    }
+  }
+
+}
+
+class DialogsThemedPresentationExampleViewController: UIViewController {
+
+  private let materialButton = MDCButton()
+  private let transitionController = MDCDialogTransitionController()
+  var containerScheme = MDCContainerScheme()
+
+  override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+    super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+    containerScheme.colorScheme = MDCSemanticColorScheme(defaults: .material201804);
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    view.backgroundColor = containerScheme.colorScheme?.backgroundColor ??
+      MDCSemanticColorScheme(defaults: .material201804).backgroundColor
+
+    materialButton.translatesAutoresizingMaskIntoConstraints = false
+    materialButton.setTitle("Custom Alert With Themed Presentation", for: .normal)
+    materialButton.setTitleColor(UIColor(white: 0.1, alpha:1), for: .normal)
+    materialButton.sizeToFit()
+    materialButton.addTarget(self, action: #selector(tap), for: .touchUpInside)
+    materialButton.applyTextTheme(withScheme: containerScheme)
+    self.view.addSubview(materialButton)
+
+    NSLayoutConstraint.activate([
+      NSLayoutConstraint(item:materialButton,
+                         attribute: .centerX,
+                         relatedBy: .equal,
+                         toItem: self.view,
+                         attribute: .centerX,
+                         multiplier: 1.0,
+                         constant: 0.0),
+      NSLayoutConstraint(item: materialButton,
+                         attribute: .centerY,
+                         relatedBy: .equal,
+                         toItem: self.view,
+                         attribute: .centerY,
+                         multiplier: 1.0,
+                         constant: 0.0)
+      ])
+  }
+
+  @objc func tap(_ sender: Any) {
+    let title = "Reset Settings"
+    let message = "Your device will reset to its default factory settings."
+    let customDialogController = CustomDialogViewController(title: title, message: message)
+    customDialogController.modalPresentationStyle = .custom;
+    customDialogController.transitioningDelegate = self.transitionController;
+
+    if let presentationController = customDialogController.mdc_dialogPresentationController {
+      presentationController.applyTheme(withScheme: containerScheme)
+      presentationController.dialogCornerRadius = customDialogController.view.layer.cornerRadius
+    }
+    present(customDialogController, animated: true, completion: nil)
+  }
+}
+
+extension DialogsThemedPresentationExampleViewController {
+
+  class func catalogMetadata() -> [String: Any] {
+    return [
+      "breadcrumbs": ["Dialogs", "Custom Dialog with Themed Presentation Style"],
```

Comment:
The name of this example is a bit long. Maybe "Dialog Presentation Controller Theming"?

Reply:
Sure. Thanks for suggestion.

Reply:
Addressed in 7c3b8a2b897c3552b19b33a3ae50a2c7d9c9ee77